File: app/src/main/java/com/example/ara/atmospher/Interfaces/OpenWeatherService.java
package com.example.ara.atmospher.Interfaces;

import com.example.ara.atmospher.WeatherData;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OpenWeatherService {

    @GET("/data/2.5/weather")
    Call<WeatherData> weatherCall(@Query("q") String cityName, @Query("appid") String API_KEY);

}

File: app/src/main/java/com/example/mahdi/atmosphere/retrofit/OpenWeatherService.kt
package com.example.mahdi.atmosphere.retrofit

import com.example.mahdi.atmosphere.models.openWeather.Current
import com.example.mahdi.atmosphere.models.openWeather.Forecast5
import com.example.mahdi.atmosphere.models.openWeather.oneCall.OneCall
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherService {
    @GET("/data/2.5/onecall?units=metric")
    suspend fun oneCall(@Query("lat") lat: Double,
                        @Query("lon") lon: Double,
                        @Query("appid") API_KEY: String): Response<OneCall>?
}